### ISSUE TITLE
Custom header image per department

### ISSUE THREAD

**Issue by Pieter1024** [jonge-democraten/website#6]

**Comment by bartromgens** [comment]
I propose the following header image structure.

A page header,
* A header consists of one or multiple images. 
* In case of a multiple image header, a random image will be selected on each page request.
* Header images are selected/uploaded from the media library. 
* Header images have a fixed display height and width.
* If possible, a correct image aspect ratio is enforced/checked. 

Is shown on a page,
* Every Page can have a header, or explicitly disable a header.
* In case no header is selected for a page, and the header has not been disabled, the parent page header will be used. 
* For top level pages, the parent is the HomePage, which contains the highest level (department) header. 

With scope and permissions,
* Headers are SiteRelated, so each Site (department) has its own headers.
* Headers can be created and modified by content managers.

**Comment by Pieter1024** [comment]
Good idea. I have two comments:

* Closing issue 37 in favor of this one is quite ambitious. The reason we had two issues is that we can live with not being able to change the headers from the backend, but we *need* headers. I hope you are certain we can get this done in time.

* I like the inheritance of headers idea. Make sure to check inheritance when the page is called, not when it is created. This makes sure that moving the page through the website structure changes the headers accordingly.

**Comment by bartromgens** [comment]
I am fairly certain we can get this done, including #37, in time. 
I plan to have a working prototype ready within a week. If we can agree on that functionality, I am sure we can get this done. If not, I will reopen #37. I mainly closed it because I got confused with the issues and want to have all discussions here. This may not be a valid reason from the feature tracking perspective though, now that I think of it.

> Make sure to check inheritance when the page is called, not when it is created.
That is definitely part of the plan. Important one. Would not dare to store information when it can be generated :)

Thanks for the feedback!

**Comment by bartromgens** [comment]
I have created the proposed customizable header image in a feature branch.

It works for the HomePage and RichTextPage. Those admin pages have the following extra fields that define the header type and the optional header image(s), 
![header_interface_example](https://cloud.githubusercontent.com/assets/2451596/6238600/be22e784-b6fc-11e4-8308-fc70276dbf74.png)

There is a problem the blogpost listings though. These are created as RichTextPage that redirect to a mezzanine view. I don't think it's possible to add a header to this view that is defined for the RichTextPage that redirects to this view. 

As I see it now, we have two options,
* Create a custom BlogListing page
* No headers on blog pages

The first would allow content managers to create a new BlogListing page, and select a blog category for the page.  
This improves the current method to add a blog page. Which is done by adding a RichTextPage with a url to the blog category, which is hidden after saving the page. This is not very obvious, transparent and user friendly. And it does not allow us to add extra options/settings to the page. 

My proposal is thus to create a BlogListing page type, derived from Page.

**Comment by bartromgens** [comment]
The proposal above has been implemented in commit 41cc6e2552b6da69b075350897cee38826101950. 

I also added a hard-coded requirement on the resolution of the page header image. The backend user will get the following field validation error in case the resolution is not correct, 
> "Image should be 610 x 290 pixels, selected image is N x N. Please resize the image."

Every HomePage and RichTextPage can have a header at the moment. It is easy to add it to other page types in the future, so I think this issue can be close.

**Comment by Pieter1024** [comment]
I don't know how to add it to other page types. Furthermore, I don't feel that that is something 'for the future', as we have other page types now as well.

For example, in the demo data the following page types have no heading yet:
* blog listings
* individual blogs
* document listings
* forms
* (and possibly others).

Could you fix this, and document how you fixed it?

**Comment by bartromgens** [comment]
Yes. It should be added to existing pages. 
I just added them to the document listing page in commit 6759b0318ca24a4358f903155fc2468e7a9e0081. 
And to the form page in commit 39f8c7c244adcac5f35fdfbc0274a870e5b9a07d.

I also added the header to the blog page in the branch for issue #70, which is not yet merged.

An example of how to add a page header image to custom page types can now be found in the design document , http://jdwebsite.readthedocs.org/en/latest/design/#page-header-image
